pkg/operator/operator.go

```diff
@@ -169,13 +179,29 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 		SecurityGroupProvider:     securityGroupProvider,
 		AMIProvider:               amiProvider,
 		AMIResolver:               amiResolver,
+		VersionProvider:           versionProvider,
 		LaunchTemplateProvider:    launchTemplateProvider,
 		PricingProvider:           pricingProvider,
 		InstanceTypesProvider:     instanceTypeProvider,
 		InstanceProvider:          instanceProvider,
 	}
 }
 
+func (op *Operator) ValidateK8sVersion(ctx context.Context) error {
+	v := lo.Must1(op.VersionProvider.Get(ctx))
+	k8sVersion := versionv1.MustParseGeneric(v)
+
+	// We will only error if the user is running karpenter on a k8s version,
+	// that is out of the range of the minK8sVersion and maxK8sVersion
+	if k8sVersion.LessThan(versionv1.MustParseGeneric(minK8sVersion)) ||
+		k8sVersion.AtLeast(versionv1.MustParseGeneric(maxK8sVersion)) &&
```

Comment:
Can we just do `|| !k8sVersion.LessThan(versionv1.MustParseGeneric(maxK8sVersion)` to get the greater than check?

Reply:
Wouldn't that be greater then or equals?

Reply:
Sorry other way. `maxK8sVersion.LessThan(k8sVersion)`